### PR TITLE
Adding OEM handler - get system time

### PR DESCRIPTION
**Pull request by AdiFogel**
Fixes nvbug https://redmine.mellanox.com/issues/3444628 Test:
fnFun = 0x2e (OEM command)
cmd = 1
data = 0x1 0x0 0x0 (iana commad number)
From BMC:
root@dpu-bmc:~# ipmitool -I ipmb raw 0x2e 1 0x1 0x0 0x0
 01 00 00 d9 4f 57 64

### CODE REVIEW

**Comment by AdiFogel** [comment]
@asmaamellanox and @odrang - 
can you please review?

Tnx

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/*
+ * mellanox_bf_mod.c
+ *
+ * Nvidia bluefield specific modules for handling BMC and MC functions.
+ *
+ * Author: MontaVista Software, Inc.
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:6]: This licensing is wrong. Please use the correct template.

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#include <OpenIPMI/serv.h>
+#include <lanserv/OpenIPMI/mcserv.h>
+#include <lanserv/bmc.h>
+
+
+#define PVERSION "1.0.1"
+#define IPMI_IANA_OEM_GET_SYSTEM_TIME_CMD	0x01
+
+/**************************************************************************
+ * Marvell OEM commands.
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:77]: Some leftovers from copy/paste. Please remove this line.

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+/**************************************************************************
+ * Module initialization
+ *************************************************************************/
+
+int ipmi_sim_module_init(sys_data_t *sys, const char *initstr_i)
+{
+    int rv;
+
+    printf("IPMI Simulator Mellanox BF module version %s\n", PVERSION);
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:109]: You dont need to print this again. It is already handled by ipmi_sim_module_print_version

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    rv = ipmi_emu_register_iana_handler(IPMI_IANA_OEM_GET_SYSTEM_TIME_CMD,
+                ipmi_get_system_time, sys);
+    if (rv) {
+        sys->log(sys, OS_ERROR, NULL,
+            "Unable to register Marvell IANA handler: %s", strerror(rv));
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:115]: copy/paste typo. Please remove the Marvell related message or fix it.

**Review by asmaamellanox — COMMENTED** [review]
please provide a proper explanation in the commit message. We dont usually just point to the redmine. Something like:
Support new Mellanox OEM command to enable the BMC to retrieve the REALTIME from the BF. The real time is retrieved via the posix function: clock_gettime.

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 1 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c:6]: where can I found the correct template?

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 3 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 4 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 2 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+
+#define PVERSION "1.0.1"
+#define IPMI_IANA_OEM_GET_SYSTEM_TIME_CMD	0x01
+
+/**************************************************************************
+ * Nvidia - Mellanox OEM commands.
+ *************************************************************************/
+
+static void ipmi_get_system_time(lmc_data_t    *mc,
```
Comment (odrang) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:80]: Please add description for the function

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 1 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c (hunk 5 above)
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c:38]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 6]
```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause
+
+/* Gigabit Ethernet driver for Mellanox BlueField SoC
+ *
+ * Copyright (C) 2020-2021 NVIDIA CORPORATION & AFFILIATES
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:5]: please update the year 
Instead of "2020-2021" change it to "2023"

Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 7]
```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause
+
+/* Gigabit Ethernet driver for Mellanox BlueField SoC
```
[the remark this answers is not in the document]
Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: missed that.... done

File: lanserv/mellanox-bf/mellanox_bf_mod.c [hunk 8]
```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0-only OR BSD-3-Clause
+
+/* OEM handlers for mlx-Open Mellanox BlueField
```
Comment (asmaamellanox) [inline lanserv/mellanox-bf/mellanox_bf_mod.c:3]: Please remove "mlx-Open" as it doesnt represent anything and we might need to upstream this file in the future

Comment (AdiFogel) [reply lanserv/mellanox-bf/mellanox_bf_mod.c]: done

**Pull request merged**